Gems/Atom/Feature/Common/Code/Source/RayTracing/RayTracingAccelerationStructurePass.cpp

```diff
@@ -60,28 +60,25 @@ namespace AZ
                 if (rayTracingFeatureProcessor->GetRevision() != m_rayTracingRevision)
                 {
                     RHI::RayTracingBufferPools& rayTracingBufferPools = rayTracingFeatureProcessor->GetBufferPools();
-                    RayTracingFeatureProcessor::MeshMap& rayTracingMeshes = rayTracingFeatureProcessor->GetMeshes();
+                    RayTracingFeatureProcessor::SubMeshVector& subMeshes = rayTracingFeatureProcessor->GetSubMeshes();
                     uint32_t rayTracingSubMeshCount = rayTracingFeatureProcessor->GetSubMeshCount();
 
                     // create the TLAS descriptor
                     RHI::RayTracingTlasDescriptor tlasDescriptor;
                     RHI::RayTracingTlasDescriptor* tlasDescriptorBuild = tlasDescriptor.Build();
 
-                    uint32_t blasIndex = 0;
-                    for (auto& rayTracingMesh : rayTracingMeshes)
+                    uint32_t instanceIndex = 0;
+                    for (auto& subMesh : subMeshes)
                     {
-                        for (auto& rayTracingSubMesh : rayTracingMesh.second.m_subMeshes)
-                        {
-                            tlasDescriptorBuild->Instance()
-                                ->InstanceID(blasIndex)
-                                ->HitGroupIndex(blasIndex)
-                                ->Blas(rayTracingSubMesh.m_blas)
-                                ->Transform(rayTracingMesh.second.m_transform)
-                                ->NonUniformScale(rayTracingMesh.second.m_nonUniformScale)
-                                ;
-                        }
-
-                        blasIndex++;
+                        tlasDescriptorBuild->Instance()
+                            ->InstanceID(instanceIndex)
+                            ->HitGroupIndex(0)
+                            ->Blas(subMesh.m_blas)
+                            ->Transform(subMesh.m_mesh->m_transform)
+                            ->NonUniformScale(subMesh.m_mesh->m_nonUniformScale)
+                            ;
+
+                        instanceIndex++;
```

Comment:
You can probably save many hits to the allocator in this loop by reserving the RayTracingTlasDescriptor::m_instances container to be the same size as the subMeshes Vector.